### PR TITLE
docs: Copilot review-inbox + category-dashboard delivery design

### PR DESCRIPTION
**Pull request by ftchvs**
## What

Adds the delivery design that backs the new Linear backlog (epics **AND-518…AND-524**, issues **AND-525…AND-559**, all related to the completed epic AND-398) for bringing [Copilot.money](https://www.copilot.money)-style transaction review + the category/budget dashboard to VaultPeek.

Doc: `docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md`

## Why it matters

The design was produced by a parallel discovery workflow (code audit + Copilot teardown + macOS 26 API refs + synthesis) plus a Linear dedup pass. Its central reframe:

- **This is wiring + polish, not a rebuild.** VaultPeek already has the review engine (`TransactionReviewInbox.evaluate`), NL categorizer, `CategoryBudgetPlanner`, server-persisted budget CRUD, and a live `ReviewInboxView`.
- **One load-bearing correctness bug:** `CategoryBudgetPlanner.netSpendByCategory` reads raw `transaction.category` and ignores user overrides / `excludedFromBudgets` / rules — so recategorizing in the inbox changes no downstream number. Fixing this (AND-518) gates everything else.
- **Category-model decision (ratified):** Option A — display-only rollups over the flat `SpendingCategory` enum, **no migration** — for v1; Option B (custom/nested categories, rollover, rebalance, splits) deferred under AND-524, consistent with the launch cutline.

## Contents

Executive summary · the category-model decision table · target experience (v1 review inbox + category dashboard) · architecture (additive, PlaidBarCore-first, no migration) · the full epic/issue/sub-issue tree with AND IDs · dependency sequencing + critical path · risks, edge cases, and the deferred v2 list.

## Notes

- Docs-only change; no code touched.
- The issues themselves are already filed in Linear; this commits the durable spec alongside them.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Two surfaces on the existing **popover-primary + detached-window** split (popover = Copilot's dashboard card; window = Copilot's full tab).
+
+- **Review Inbox** ("triage to zero"): bulk "Mark N reviewed" + blast radius, date-group sections, colored icon+text category pill, inline rule prompt, menu-bar count badge; detached window gets a multi-select `Table`. Keyboard speed-review (`a/i/c/t/r/m`, arrows, ⌘Z) already ships — a native-macOS advantage, kept.
+- **Category Dashboard** ("where money went vs. plan"): a new center-column card (donut + top groups) → detached full window (monthly-history `BarMark` + dashed `RuleMark` budget line + flat SPENT/BUDGET/LEFT `Table`). Built as Option-A rollups.
+
+**Cut from v1 (YAGNI):** splits, rollover, rebalance, custom/renamed categories, per-month budgets, server-synced review, swipe gestures. All in Epic 7.
+
+## 4. Architecture (Option A — additive, no migration)
+
+- **New (PlaidBarCore):** `CategoryGroup` enum + `SpendingCategory.group`; `CategoryDashboardBuilder` (pure rollup, `asOf:`/`Calendar` injected, `Sendable`); extracted `EffectiveCategoryResolver` (lifted from `TransactionReviewInbox.resolveCategory:380-404`).
```
[the remark this answers is not in the document]
Comment (ftchvs) [reply docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md:40]: Fixed in f49761c. The §4 `EffectiveCategoryResolver` now has two modes: the inbox keeps display mode (incl. NL fallback), while budget/dashboard **aggregation uses a persisted-only mode** (user override → rules → raw Plaid → `.other`) with NL display suggestions **excluded** — so an unreviewed NL guess never moves budget/category totals, and approving in the inbox stays the one place numbers change.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- **Changed (the fix):** `CategoryBudgetPlanner.netSpendByCategory` gains optional `metadata`/`rules` params (default nil → existing tests pass); skips excluded/transfer rows; resolves effective category. `AppState.categoryBudgetPresentation:1251` passes live metadata/rules.
```
[the remark this answers is not in the document]
Comment (ftchvs) [reply docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md:41]: Fixed in f49761c. §4 now requires invalidating the budget-presentation cache when `transactionReviewMetadata`/`transactionRules` change (not only `transactions`/`categoryBudgets` as today at `AppState.swift:69-82,144-145`), so an approve/recategorize/new-rule no longer leaves the dashboard stale.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+Phase 3 (budget edit): AND-540 (needs 526) → 541 (needs 539+540), 542 (needs 540)
+Phase 4 (QA):          AND-544 (needs 543+526) ; AND-545 (needs 539+540+528)
+```
+
+**Critical path:** AND-525 → 526 → 536 → 538 → 539 → 541. `blockedBy` links are wired in Linear.
+
+## 7. Risks, edge cases, future
+
+- **Risks:** scope creep into a budgeting suite (Option A hard-caps v1); override-awareness regression (additive nil-default params keep tests green); `BudgetStore.allBudgets` silently drops unmapped category rows (`BudgetStore.swift:25-28`) — v1 never changes rawValues, guarded; menu-bar badge / glass constraints (custom `NSStatusItem`, no glass-on-glass); macOS-26 API availability (`.sectionActions` gated by `#available`).
+- **Edge cases:** pending→posted (already handled, de-dupe in dashboard); transfers/credit-card payments (exclude via override); NL-suggested vs persisted category (display-only, stays `.uncategorized` until approved); over-at-leaf-but-under-at-group (independent status); multi-account aggregation vs account-filter; empty/first-run (suggestions only, no false "over"); refunds (signed-amount netting).
```
[the remark this answers is not in the document]
Comment (ftchvs) [reply docs/superpowers/specs/2026-06-18-copilot-review-category-dashboard-design.md:74]: Fixed in f49761c. Both §4 and the §7 edge-cases now call out that the aggregation resolver must carry review metadata saved under `pendingTransactionId` into the posted replacement (mirroring `TransactionReview.swift:207-222`), not just `resolveCategory:380-404` — so a pending-phase category/transfer decision survives the re-post under a new id.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
